src/encoded/submit.py

```diff
@@ -760,18 +783,252 @@ def create_json_out(self):
+        if the proband needs to be changed (e.g. the family has 2 affected siblings). In these cases
+        we want to update the family history for both families, so we will look up the family ID and patch
+        all families with the ID.
+        """
+        family_metadata = {}
+        for alias, item in self.individuals.items():
+            family_metadata.setdefault(item['family_id'],
+                                       {'family_id': item['family_id'], 'members': []})
+            family_metadata[item['family_id']]['members'].append(alias)
+            if item.get('proband', False):
+                if 'proband' not in family_metadata[item['family_id']]:
+                    family_metadata[item['family_id']]['proband'] = alias
+                    family_metadata[item['family_id']]['aliases'] = [
+                        self.project + ':family-' + item['individual_id']
+                    ]
+                else:
+                    msg = ('More than one proband indicated for family {}. Please indicate'
+                           ' a single proband in the spreadsheet and resubmit'.format(item['family_id']))
+                    self.errors.append(msg)
+            del item['family_id']
+        final_family_dict = {}
+        for key, value in family_metadata.items():
+            try:
+                family_matches = self.virtualapp.get(f'/search/?type=Family&family_id={key}')
+            except Exception:
+                # if family not in DB, create a new one
+                # first make sure a proband is indicated for a family if its not already in DB
+                if not value.get('proband'):
+                    msg = 'No proband indicated for family {}. Please edit and resubmit'.format(value['family_id'])
+                    self.errors.append(msg)
+                else:
+                    final_family_dict[value['aliases'][0]] = value
+            else:
+                for match in family_matches.json['@graph']:
+                    # value.setdefault('aliases', [])
+                    # value['aliases'].extend(match.get('aliases', []))
+                    final_family_dict[match['@id']] = value
+                    #final_family_dict[match['@id']]['aliases'] = match.get('aliases', [])
+                    if value.get('proband'):
+                        phenotypes = list(set([item['phenotypic_feature'] for item in
+                                      self.individuals[value['proband']].get('phenotypic_features', [])]))
+                        # Add other family member phenotypes if proband phenotypes < 4
+                        if len(phenotypes) < 4:
+                            for member in value['members']:
+                                if member != value['proband']:
+                                    member_phenotypes = [item['phenotypic_feature'] for item in
+                                                         self.individuals[member].get('phenotypic_features', [])]
```

Comment:
I barely like having an inline `for` on the same line as the body it's using, and often just say the `for` should just be on the next line anyway. But _especially_ if you have to break things, do it before the `for` not within that clause. It's OK to break in other places as well. I like to put parens around the thing that will break even though the fact that you're in a square-bracket context already might not require that.
```suggestion
                                    member_phenotypes = [item['phenotypic_feature']
                                                         for item in (self.individuals[member]
                                                                      .get('phenotypic_features', []))]
```